core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -266,10 +266,12 @@ private[spark] class ExecutorAllocationManager(
       // executors and inform the cluster manager to cancel the extra pending requests
       val oldNumExecutorsTarget = numExecutorsTarget
       numExecutorsTarget = math.max(maxNeeded, minNumExecutors)
-      client.requestTotalExecutors(numExecutorsTarget)
-      numExecutorsToAdd = 1
-      logInfo(s"Lowering target number of executors to $numExecutorsTarget because " +
-        s"not all requests are actually needed (previously $oldNumExecutorsTarget)")
+      if (numExecutorsTarget < oldNumExecutorsTarget) {
```

Comment:
basically, if the contract is "we start `numExecutorsToAdd` at 1, and every time we decrease we reset it to 1", then I think we get everything we want, no?

obvs it's a nit to pick, just want to be sure we're all agreed about the semantics and more importantly that I understand the code I'm modifying :)